### PR TITLE
Publish tagged versions to RubyGems

### PR DESCRIPTION
**Pull request by sgerrand**
💁 These changes:
* add a CircleCI job configuration
* automatically build and push the gem to RubyGems when a git tag is pushed.
* cleanup the gemspec
* attach the correct license

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - save_cache:
+          key: capistrano-ext-superusers-{{ checksum "capistrano-ext-superusers.gemspec" }}
+          paths:
+            - vendor/bundle
+      - run: bundle exec rspec spec --format progress
+      - deploy:
+          name: Build gem and push to RubyGems
+          command: |
+            if [ ! -z "$CIRCLE_TAG" ]; then
+              gem build $CIRCLE_PROJECT_REPONAME.gemspec
```
Comment (Ninju) [inline .circleci/config.yml]: **curious** Could `$CIRCLE_PROJECT_REPONAME` ever be different to `capistrano-ext-superusers`?

I am wondering why we've hardcoded `capistrano-ext-superusers` elsewhere. Is there any difference? Does it matter? 🤷‍♂️

File: capistrano-ext-superusers.gemspec [hunk 2]
```diff
@@ -6,11 +6,14 @@ Gem::Specification.new do |spec|
...
-  spec.description = 'Capistrano extension to run sensible userage'
-  spec.license = 'Simplified BSD'
+  spec.authors = ['Funding Circle Engineering']
+  spec.email = ['engineering+capistrano-ext-superusers@fundingcircle.com']
+  spec.summary = 'Run Capistrano commands as a superuser'
+  spec.description = 'Capistrano extension to run commands as an unprivileged user in a sensible manner'
+  spec.homepage = 'https://github.com/FundingCircle/capistrano-ext-superusers'
+  spec.license = 'BSD-3-Clause'
+
+  spec.files = `git ls-files -z -- ./* ':(exclude)spec/*'`.split("\x0")
```
[the remark this answers is not in the document]
Comment (sgerrand) [reply capistrano-ext-superusers.gemspec]: 😊  Shall remove!

File: .circleci/config.yml (hunk 1 above)
Comment (sgerrand) [reply .circleci/config.yml]: > Could `$CIRCLE_PROJECT_REPONAME` ever be different to `capistrano-ext-superusers`?

Not unless the repository is renamed.

Comment (sgerrand) [reply .circleci/config.yml]: > I am wondering why we've hardcoded `capistrano-ext-superusers` elsewhere. Is there any difference? Does it matter?

I'd love to use `$CIRCLE_PROJECT_REPONAME` in the CircleCI job configuration - e.g. for the caching commands - but they're not available for use there. 😞

Comment (Ninju) [reply .circleci/config.yml]: That's what I thought. :smile:

TIL. Thanks!

File: capistrano-ext-superusers.gemspec (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgerrand) [reply capistrano-ext-superusers.gemspec]: Fixed in 916a5de. ♻️

[the remark this answers is not in the document]
Comment (Ninju) [reply capistrano-ext-superusers.gemspec]: Are you missing some commits? There were 5, and now there are 2. The `.circleci/config.yml` is no longer in the diff.

[the remark this answers is not in the document]
Comment (sgerrand) [reply capistrano-ext-superusers.gemspec]: Thanks for spotting that. They have re-emerged. ✨

**Pull request merged**